streamlit_web.py

```diff
@@ -283,6 +300,24 @@ def overall_freq(freq_range):
         )
     )
 
+    responses_end = len(main_df.columns) - 3
+    responses_df = main_df[main_df.columns[1:responses_end]]
+    responses_df.replace("", "NA")
+
+    # concatenate all words into normalized string and make into wordcloud
+    words = az.concatenate(responses_df)
```

Comment:
See the previous `tokenize()` comment. Also, you might be able to get this from the `main_df` even.

Reply:
We can't just use base `main_df`. It contains too many things that aren't the actual individual's questions and their responses, which we resolve by making `responses_df` out of the `stu_assignment` pulled from the `main_df`, and the input for the wordcloud needs to be one concatenated string of all of the responses instead of separate strings in a dataframe, which we use concatenate to resolve. 
As said previously, tokenize couldn't be used instead because tokenize returns a list of tokens instead of a concatenated string. If it was used, it would have to be used very redundantly by passing in each response from each individual, as it only accepts one string at a time, and then making each list returned into a string and concatenating all of those strings, which doesn't seem like the cleanest implementation.